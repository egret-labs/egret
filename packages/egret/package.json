{
  "name": "@egret/egret",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "jest"
  },
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "@types/jest": "~26.0.19",
    "jest": "~26.6.3",
    "ts-jest": "~26.4.4",
    "typescript": "3.9.9",
    "jest-webgl-canvas-mock": "~0.2.3",
    "webgl-mock": "~0.1.7",
    "@types/node": "~14.14.33"
  },
  "jest": {
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js"
    ],
    "setupFiles": [
      "jest-webgl-canvas-mock"
    ],
    "transform": {
      "\\.(ts|tsx)$": "ts-jest"
    },
    "testRegex": "/test/.*.spec\\.(ts|tsx)$"
  }
}