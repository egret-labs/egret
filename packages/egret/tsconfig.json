{
    "compilerOptions": {
        "target": "es5",
        "sourceMap": false,
        "declaration": true,
        "allowUnreachableCode": true,
        "newLine": "LF",
        "lib": [
            "dom",
            "es2015"
        ]
    },
    "exclude": [
        "node_modules"
    ]
}