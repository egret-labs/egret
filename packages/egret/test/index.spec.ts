import 'jest-webgl-canvas-mock';
global.egret = {};
// console.log(add);
const canvas = document.createElement('canvas');
const img = document.createElement('img');
img.src = '111.png';
const gl = canvas.getContext('webgl');
require('./libs/modules/egret/egret.js');

// const ctx = canvas.getContext('2d');

class VertexBuffer {

    private gl: any;
    type: any;
    private mode: any;
    private buffer: any;

    constructor(gl: any, options?: any) {
        options = options || {};
        this.gl = gl;
        this.type = (options.type !== undefined) ? options.type : gl.FLOAT;
        this.mode = (options.mode !== undefined) ? options.mode : gl.TRIANGLES;
        this.buffer = gl.createBuffer();
        gl.bindBuffer(gl.ARRAY_BUFFER, this.buffer);
        gl.bufferData(gl.ARRAY_BUFFER, options.data || options.size, gl.STATIC_DRAW);
        gl.bindBuffer(gl.ARRAY_BUFFER, null);
    }
}

describe('VertexBuffer', function () {
    describe('#constructor()', function () {
        // it('should default type to gl.FLOAT', function () {
        //     const vb = new VertexBuffer(gl);

        //     console.log(vb.type === gl.FLOAT);
        // });
        // it('should default mode to gl.TRIANGLES', function () {
        //     const vb = new VertexBuffer(gl);
        //     console.log(vb.type === gl.TRIANGLES);
        // });
        it('ddd', () => {
            // ctx.drawImage(img, 100, 100);
            // // ctx.setTransform(2, 2, 2, 2, 100, 100);
            // // ctx.fillText('helloworld', 0, 20);

            // const dc = ctx.__getDrawCalls();
            // expect(dc).toMatchSnapshot();
        });
    });
});