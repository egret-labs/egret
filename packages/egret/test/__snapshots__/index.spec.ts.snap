// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VertexBuffer #constructor() ddd 1`] = `
Array [
  Object {
    "props": Object {
      "dHeight": 0,
      "dWidth": 0,
      "dx": 100,
      "dy": 100,
      "img": <img
        src="111.png"
      />,
      "sHeight": 0,
      "sWidth": 0,
      "sx": 0,
      "sy": 0,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "drawImage",
  },
]
`;
